client/components/language-picker/modal.jsx

```diff
@@ -417,7 +415,18 @@ export class LanguagePickerModal extends PureComponent {
 		this.handleClose();
 	};
 
-	handleClose = () => {
+	recordLanguagePickerEvent = ( isClosingWithoutSelection ) => {
+		this.props.recordTracksEvent( 'calypso_languagepicker_closed', {
+			searched: false !== this.state.search,
+			is_closing_without_selection: isClosingWithoutSelection,
+			did_select_language:
+				! isClosingWithoutSelection &&
```

Comment:
I'd keep `! isClosingWithoutSelection &&` away from the `did_select_language` logic for 2 reasons:

* In my understanding, `did_select_language` would be useful to know whether the language was changed at some point (even if it wasn't saved). However, if we consider this happened only if `isClosingWithoutSelection` is false, then we'll skip all the instances where the user selected a language but didn't click the "Select Language" button to save it. I don't have a strong feeling against this behavior if you intended this to work like that.
* This data is already available in the `is_closing_without_selection` event prop.

Additionally, if you'd like us to differentiate between "language selection" (where the change is saved) and "language choosing" (where data isn't saved), we can also reflect that in the event prop naming.